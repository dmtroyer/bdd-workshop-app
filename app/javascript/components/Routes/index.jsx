import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Game from '../Game';
import About from '../About';
import Profile from '../Profile';
import SignIn from '../SignIn';
import Register from '../Register';
import MainContainer from '../MainContainer';
import WarGamesIntro from '../WarGamesIntro';

const Routes = (props) => (
  <MainContainer>
    <Switch>
      <Route exact path="/" component={WarGamesIntro} />
      <Route exact path="/register" render={(routeProps) => <Register {...props} {...routeProps} />} />
      <Route exact path="/sign_in" render={(routeProps) => <SignIn {...props} {...routeProps} />} />
      <Route exact path="/profile" render={(routeProps) => <Profile {...props} {...routeProps} />} />
      <Route exact path="/game" component={Game} />
      <Route exact path="/about" component={About} />
    </Switch>
  </MainContainer>
);

export default Routes;
