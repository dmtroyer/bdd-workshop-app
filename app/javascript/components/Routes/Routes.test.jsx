import React from 'react';
import { shallow } from 'enzyme';
import Routes from '.';

describe('Routes', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Routes />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
