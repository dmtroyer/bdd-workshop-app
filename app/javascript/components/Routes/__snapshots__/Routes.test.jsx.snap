// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders correctly 1`] = `
<MainContainer>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/register"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/sign_in"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/profile"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/game"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
  </Switch>
</MainContainer>
`;
